pkg/backend/filestate/backend_test.go

```diff
@@ -473,3 +475,27 @@ func TestHtmlEscaping(t *testing.T) {
 	state := string(bytes)
 	assert.Contains(t, state, "<html@tags>")
 }
+
+func TestLocalBackendRejectsStackInitOptions(t *testing.T) {
+	t.Parallel()
+	// Here, we provide options that illegally specify a team on a
+	// backend that does not support teams. We expect this to create
+	// an error later when we call CreateStack.
+	illegalOptions := backend.NewStandardCreateStackOpts([]string{"red-team"})
+
+	// • Create a mock local backend
+	tmpDir := t.TempDir()
+	dirURI := fmt.Sprintf("file://%s", filepath.ToSlash(tmpDir))
+	local, err := New(context.Background(), cmdutil.Diag(), dirURI, nil)
+	assert.NoError(t, err)
+	ctx := context.Background()
+
+	// • Simulate `pulumi stack init`, passing non-nil init options
+	fakeStackRef, err := local.ParseStackReference("foobar")
+	assert.NoError(t, err)
+	assert.Panics(t, func() {
+		// • Expect an error.
+		_, err := local.CreateStack(ctx, fakeStackRef, "", illegalOptions)
+		assert.NoError(t, err)
```

Comment:
I resolved this comment; I should have indicated that the assertion on line 499 should be unreachable.

Reply:
Yeah, I think the comment should then be "This will panic" or similar (implied by the function).
You can also ignore the err return:

```go
		_, _ = local.CreateStack(ctx, fakeStackRef, "", illegalOptions)
		// err ignored because CreateStack will panic
```

Or, you can mark the test failed if it reaches the following line:

```go
		_, err := local.CreateStack(ctx, fakeStackRef, "", illegalOptions)
		require.Fail(t, "unreachable", "got error: %v", err)
```